neural_compressor/adaptor/pytorch.py

```diff
@@ -4434,6 +4457,192 @@ def calculate_op_sensitivity(self, model, dataloader, tune_cfg, output_op_names,
         return ordered_ops
 
 
+@adaptor_registry
+class PyTorchWeightOnlyAdaptor(TemplateAdaptor):
+    """Adaptor of PyTorch framework, all PyTorch API is in this class.
+
+    Args:
+        framework_specific_info (dict): dictionary of tuning configure from yaml file.
+    """
+    def __init__(self, framework_specific_info):
+        super(PyTorchWeightOnlyAdaptor, self).__init__(framework_specific_info)
+        self.tune_cfg = None
+        if self.device == "cpu":
+            query_config_file = "pytorch_cpu.yaml"
+        else:  # pragma: no cover
+            assert False, "Unsupport this device {}".format(self.device)
+        self.query_handler = PyTorchQuery(
+            local_config_file=os.path.join(os.path.dirname(__file__), query_config_file))
+
+        self.white_list = [torch.nn.Linear, torch.nn.Conv2d]
+        # Contains parameters for algorithms such as AWQ, GPTQ, etc.
+        self.recipes = framework_specific_info['recipes']
+        self.optype_statistics = None
+
+    @dump_elapsed_time("Pass quantize model")
+    def quantize(self, tune_cfg, model, dataloader, q_func=None):
+        """Execute the quantize process on the specified model.
+
+        Args:
+            tune_cfg (dict): quantization config.
+            model (object): model need to do quantization.
+            dataloader (object): calibration dataset.
+            q_func (objext, optional): training function for quantization aware training mode.
+
+        Returns:
+            (object): quantized model
+        """
+        assert isinstance(model._model, torch.nn.Module), \
+               "The model passed in is not the instance of torch.nn.Module"
+        if self.performance_only:
+            q_model = model
+        else:
+            try:
+                q_model = copy.deepcopy(model)
+            except Exception as e:  # pragma: no cover
+                logger.warning("Fail to deep copy the model due to {}, inplace is used now.".format(
+                    repr(e)))
+                q_model = model
+
+        # For tensorboard display
+        self.tune_cfg = tune_cfg
+        self.tune_cfg["approach"] = self.approach
+        self.tune_cfg["framework"] = "pytorch"
+        assert self.approach=='post_training_weight_only', "Please make sure the approach is weight_only"
+
+        q_model._model = self.RTN_quantize(q_model._model, tune_cfg)
+        q_model._model = self.GPTQ_quantize(q_model._model, tune_cfg, dataloader)
+        q_model._model = self.AWQ_quantize(q_model._model, tune_cfg, dataloader)
```

Comment:
why trying three algos?

Reply:
This is the design to combine different algos, the quantization is controlled by the tune_cfg.
For example, if there is no GPTQ in tune_cfg, we will return the input model.
The order can be changed with the best recipe we have, I will add logs in each quantize function to show the order for users.

Reply:
If user specifies multiple algos, then we can do it by internal order. However, I don't see any logics to check what algos should be used.

Reply:
It's WIP, and will be inside the self.gptq_quantize function.